src/Cropper.tsx

```diff
@@ -742,7 +742,7 @@ class Cropper extends React.Component<CropperProps, State> {
       classes: { containerClassName, cropAreaClassName, mediaClassName },
     } = this.props
 
-    const objectFit = this.state.mediaObjectFit
+    const objectFit = this.state.mediaObjectFit ?? this.getObjectFit()
```

Comment:
Could we instead add the following code inside the `componentDidMount()`:
```ts
const objectFit = this.getObjectFit()
if (objectFit !== this.state.mediaObjectFit) {
  this.setState({ mediaObjectFit: objectFit }, this.computeSizes)
}
```

I think it would fit better with the current logic of having it stored in the state.

Reply:
I just tried it and I got the following:

Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

This probably happened because calling this.computeSizes this early doesn't work. The media must be loaded because this function is called. It calculated a zoom level of NaN and everything broke from there.

I also tried to remove the "this.computeSizes" callback and this cleared the error but the initialCroppedAreaPixels didn't work. I was back to the previous issue.


I don't have time to troubleshoot further.

Reply:
Alright, let's call this good enough then :)